src/control/common/env_utils_test.go

```diff
@@ -0,0 +1,175 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package common
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func resetEnv(t *testing.T) func() {
+	t.Helper()
+
+	startEnv := os.Environ()
+	return func() {
+		os.Clearenv()
+		setEnv(t, startEnv)
+	}
+}
+
+func setEnv(t *testing.T, env []string) {
+	t.Helper()
+
+	for _, keyVal := range env {
+		fields := strings.SplitN(keyVal, "=", 2)
+		if len(fields) != 2 {
+			t.Fatalf("malformed env keyVal %q", keyVal)
+		}
+		if err := os.Setenv(fields[0], fields[1]); err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestScrubEnvironment(t *testing.T) {
+	defaultSetup := []string{
+		"FOO=bar", "baz=quux", "COW=QUACK", "ANSWER=42",
+	}
+	for name, tc := range map[string]struct {
+		setup     []string
+		list      []string
+		whitelist bool
+		expected  []string
+	}{
+		"nil blacklist": {},
+		"empty blacklist": {
+			list: []string{},
+		},
+		"blacklist": {
+			list:     []string{"COW", "baz"},
+			expected: []string{"FOO=bar", "ANSWER=42"},
+		},
+		"nil whitelist": {
+			whitelist: true,
+			expected:  []string{},
+		},
+		"empty whitelist": {
+			list:      []string{},
+			whitelist: true,
+			expected:  []string{},
+		},
+		"whitelist": {
+			list:      []string{"ANSWER"},
+			whitelist: true,
+			expected:  []string{"ANSWER=42"},
+		},
```

Comment:
Wouldn't block on this, but not crazy about having the whitelist and blacklist tests in the same test function, now that they are different function calls.